File: integration-tests/java-selenium/java-selenium-test/src/main/java/com/gu/fronts/integration/test/page/HeaderPage.java
package com.gu.fronts.integration.test.page;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

import com.gu.fronts.integration.test.fw.selenium.FindByTestAttribute;
import com.gu.fronts.integration.test.page.util.AbstractParentPage;

public class HeaderPage extends AbstractParentPage {

    @FindByTestAttribute(using = "logo")
    private WebElement logo;

    public HeaderPage(WebDriver webDriver) {
        super(webDriver);
    }

    public HeaderPage isDisplayed() {
        super.exists(logo);
        return this;
    }

    public NetworkFrontPage clickLogo() {
        logo.click();
        return loadPage(NetworkFrontPage.class);
    }

    public Editions editions() {
        return pageFactory.initPage(webDriver, Editions.class);
    }
}

File: integration-tests/scala-selenium/app/com/gu/fronts/integration/test/page/common/HeaderPage.scala
package com.gu.fronts.integration.test.page.common

import org.openqa.selenium.{ WebDriver, WebElement }

import com.gu.fronts.integration.test.fw.selenium.FindByTestAttribute
import com.gu.fronts.integration.test.page.nwfront.NetworkFrontPage
import com.gu.fronts.integration.test.page.util.FrontsParentPage
import com.gu.fronts.integration.test.page.util.PageElementHelper.existsAndDisplayed

class HeaderPage(webDriver: WebDriver) extends FrontsParentPage(webDriver) {

  @FindByTestAttribute(using = "logo")
  private var logo: WebElement = _

  def isDisplayed(): Boolean = {
    existsAndDisplayed(logo)
  }

  def clickLogo(): NetworkFrontPage = {
    logo.click()
    loadPage(classOf[NetworkFrontPage])
  }

  def editions(): Editions = {
    pageFactory.initPage(webDriver, classOf[Editions])
  }
}